partialWaveFit/pwaLikelihood.cc

```diff
@@ -1101,33 +1125,86 @@ pwaLikelihood<complexT>::addAmplitude(const amplitudeMetadata& meta)
 	// get normalization
 	const complexT normInt = _normMatrix[refl][waveIndex][refl][waveIndex];
 
-	// connect tree leaf
-	amplitudeTreeLeaf* ampTreeLeaf = 0;
-	meta.amplitudeTree()->SetBranchAddress(amplitudeMetadata::amplitudeLeafName.c_str(), &ampTreeLeaf);
-	vector<complexT> amps;
-	for (long int eventIndex = 0; eventIndex < meta.amplitudeTree()->GetEntriesFast(); ++eventIndex) {
-		meta.amplitudeTree()->GetEntry(eventIndex);
-		if (!ampTreeLeaf) {
-			printWarn << "null pointer to amplitude leaf for event " << eventIndex << ". "
-			          << "skipping." << endl;
-			continue;
+	bool onTheFlyBinning = false;
+	if (_eventFileProperties.size() > 0) {
+		onTheFlyBinning = true;
+	}
+	size_t totalEvents = 0;
+	for (size_t nMeta = 0; nMeta < metas.size(); ++nMeta) {
+		if (onTheFlyBinning) {
+			std::vector<rpwa::eventMetadata> evtMetas = metas[nMeta]->eventMetadata();
+			for (size_t nEvt = 0; nEvt < evtMetas.size(); ++nEvt) {
+				totalEvents += _eventFileProperties[evtMetas[nEvt].contentHash()].second.size();
+			}
+		} else {
+			totalEvents += metas[nMeta]->amplitudeTree()->GetEntriesFast();
 		}
-		assert(ampTreeLeaf->nmbIncohSubAmps() == 1);
-		complexT amp(ampTreeLeaf->incohSubAmp(0).real(), ampTreeLeaf->incohSubAmp(0).imag());
-		if (_useNormalizedAmps) {  // normalize data, if option is switched on
-			if (normInt == (value_type)0. && amp != (value_type)0.) {
-				printErr << "normalization integral for wave '" << waveName << "' is zero, but the amplitude is not. Aborting...";
-				return false;
+	}
+	vector<complexT> amps(totalEvents);
+
+	size_t eventCount = 0; // Running count for event number over all single files
+	size_t runningEventFileIndex = 0;
+	for (size_t nMeta = 0; nMeta < metas.size(); ++nMeta) {
+		const rpwa::amplitudeMetadata* meta = metas[nMeta];
+		// connect tree leaf
+		amplitudeTreeLeaf* ampTreeLeaf = 0;
+		meta->amplitudeTree()->SetBranchAddress(amplitudeMetadata::amplitudeLeafName.c_str(), &ampTreeLeaf);
+
+		size_t skipEvents  = 0;
+		size_t nmbEvtFiles = 1;
+		if (onTheFlyBinning) {
+			nmbEvtFiles = meta->eventMetadata().size();
+		}
+		for (size_t iEvt = 0; iEvt < nmbEvtFiles; ++iEvt) {
+			std::string evtHash = "";
+			if (onTheFlyBinning) {
+				evtHash = meta->eventMetadata()[iEvt].contentHash();
+				if (_eventFileHashOrder.size() < runningEventFileIndex+1){
```

Comment:
What if I add more amplitudes for the second wave than for the first? That would not be caught by this, would it? It should be, should it not?